File: frontend/types/skill-gap.ts
export type SkillGapPriority = "high" | "medium" | "low";

export interface MissingSkillGapItem {
  skill: string;
  priority: SkillGapPriority;
  source: string;
  why_it_matters: string;
}

export interface LearningRoadmapStep {
  skill: string;
  rationale: string;
}

export interface SkillGapResult {
  job_match_id: string;
  job_description_id: string;
  resume_id: string;
  resume_version_id: string | null;
  cached: boolean;
  target_role: string | null;
  company: string | null;
  skill_coverage_percent: number;
  coverage_meta: {
    required_count?: number;
    preferred_count?: number;
    matched_weight?: number;
    total_weight?: number;
    formula?: string;
  };
  matched_skills: string[];
  missing_skills: MissingSkillGapItem[];
  learning_roadmap: LearningRoadmapStep[];
  match_score: number;
  analyzed_at: string | null;
  prompt_version: string;
}

export const PRIORITY_LABELS: Record<SkillGapPriority, string> = {
  high: "High priority",
  medium: "Medium priority",
  low: "Low priority",
};

File: backend/app/schemas/skill_gap.py
"""Skill gap API schemas."""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import Field

from app.schemas.common import BaseSchema


class MissingSkillItem(BaseSchema):
    skill: str
    priority: Literal["high", "medium", "low"]
    source: str
    why_it_matters: str


class RoadmapStepRead(BaseSchema):
    skill: str
    rationale: str


class SkillGapResponse(BaseSchema):
    job_match_id: uuid.UUID
    job_description_id: uuid.UUID
    resume_id: uuid.UUID
    resume_version_id: uuid.UUID | None = None
    cached: bool
    target_role: str | None = None
    company: str | None = None
    skill_coverage_percent: float
    coverage_meta: dict[str, Any] = Field(default_factory=dict)
    matched_skills: list[str] = Field(default_factory=list)
    missing_skills: list[MissingSkillItem] = Field(default_factory=list)
    learning_roadmap: list[RoadmapStepRead] = Field(default_factory=list)
    match_score: int
    analyzed_at: datetime | None = None
    prompt_version: str = "skill_gap_v1"
